thavalon-server/src/game/mod.rs

```diff
@@ -0,0 +1,247 @@
+//! THavalon game logic
+
+use std::collections::HashMap;
+use std::fmt;
+use std::ops::Index;
+
+use rand::prelude::*;
+
+pub mod builder;
+mod engine;
+mod interactions;
+mod messages;
+mod role;
+
+pub use self::messages::*;
+pub use self::role::*;
+
+/// Key for identifying a player in the game. Cheaper to copy and move around than a String
+pub type PlayerId = usize;
+
+/// A mission number (from 1 to 5)
+pub type MissionNumber = u8;
+
+/// A proposal number (starts at 0)
+pub type ProposalNumber = u8;
+
+#[derive(Debug, Clone)]
+pub struct Game {
+    players: Players,
+    info: HashMap<PlayerId, String>,
+    proposal_order: Vec<PlayerId>,
+    spec: &'static GameSpec,
```

Comment:
nit: Should GameSpec be defined before Game? Since both are in same file, I feel like since Game uses GameSpec then GameSpec should be defined first.